### PR TITLE
Fix two security issues 

### PR DESCRIPTION
**Pull request by YusukeJustinNakajima**
## Description

Fixed below functions
- openapi.py 
   - Fix yaml.load function argument to prevent arbitrary code execution
- json.py
   - Added character length limitation to fix ReDoS issue
 
## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (does not change functionality, e.g. code style improvements, linting)
- [ ] Documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Please delete options that are not relevant.

- [ ] Unit Test
- [ ] Test Script (please provide)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

## Maintainer Checklist

- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] Made sure Checks passed

### CODE REVIEW

**Comment by YusukeJustinNakajima** [comment]
@deshraj 
I have fixed the security issue that I communicated to you via email. It would be very helpful if you could give priority to this issue.

**Review by deshraj — APPROVED** [review]

File: embedchain/loaders/json.py [hunk 1]
```diff
@@ -49,6 +49,9 @@ def _check_content(content):
                         Check - `https://docs.embedchain.ai/data-sources/json`"
             )
 
+        if len(content) > MAX_CONTENT_LENGTH:
+            raise ValueError(f"The content is too long. Maximum allowed length is {MAX_CONTENT_LENGTH} characters.")
```
Comment (deshraj) [inline embedchain/loaders/json.py:53]: Actually just realized that this is quite limiting. We don't want to limit our users from loading a large json. Can you please solve this a better way or remove it?

**Review by deshraj — CHANGES_REQUESTED** [review]

File: embedchain/loaders/json.py (hunk 1 above)
Comment (YusukeJustinNakajima) [reply embedchain/loaders/json.py]: @deshraj 
According to your comment, I removed string length limitation. Instead, regular expressions have been modified to mitigate the effects of the ReDoS vulnerability. A string with a length of 30000 took about **20** seconds, but after the modification, the time is **0.003** seconds for a string of the same length.

File: embedchain/loaders/json.py [hunk 2]
```diff
@@ -48,7 +46,7 @@ def _check_content(content):
                     `json.dump(data, indent=0)` and add the stringified JSON. \
                         Check - `https://docs.embedchain.ai/data-sources/json`"
             )
-
+
```
Comment (deshraj) [inline embedchain/loaders/json.py:49]:
```suggestion

```

File: embedchain/loaders/json.py (hunk 1 above)
Comment (YusukeJustinNakajima) [reply embedchain/loaders/json.py]: @deshraj 
Thanks for checking it out. To be honest, I am new to this area. 
I am sure I could have formatted it, but could you please check?

**Review by deshraj — APPROVED** [review]

**Pull request merged**